src/bokeh/models/annotations/legends.py

```diff
@@ -564,8 +574,24 @@ def _check_non_matching_scale_bar_unit(self):
     Whether the scale bar should be oriented horizontally or vertically.
     """)
 
-    location = Enum(Anchor, default="top_right", help="""
-    Location anchor for positioning scale bar.
+    # TODO rename to position and deprecate
```

Comment:
in common usage "location" and "position" are more or less synonymous. What precise distinction do you want to draw between these two words, that we need to rename?

Reply:
I know that they are practically synonymous, but in our API we have two quite different usages for `location`. One is relative placement based on `Location` enum, used in positioning toolbar in a plot or a grid plot, labels and titles in various models (including `ScaleBar`), etc. The other is placement within a container, used in `ScaleBar`, `Legend`, etc. Thus it would be nice to use different term for different applications, given there's a choice. Also, it's more a thought rather than something actionable in the near future.